jetty-security/src/main/java/org/eclipse/jetty/security/RolePrincipal.java

```diff
@@ -0,0 +1,48 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.security;
+
+import java.io.Serializable;
+import java.security.Principal;
+import javax.security.auth.Subject;
+
+/**
+ * RolePrincipal
+ */
+public class RolePrincipal implements Principal, Serializable
+{
+    private static final long serialVersionUID = 2998397924051854402L;
+    private final String _roleName;
```

Comment:
If this is just a String, then why not just have `String role` and `List<String> roles` etc?

Reply:
ah because it was there before as an inner class!